src/Android/Xamarin.Android/Samples/Network Analysis/ClosestFacility/ClosestFacility.cs

```diff
@@ -0,0 +1,207 @@
+            foreach (Facility facility in _facilities)
+            {
+                _facilityGraphicsOverlay.Graphics.Add(new Graphic(facility.Geometry, _facilitySymbol));
+            }
+
+            // Add each graphics overlay to MyMapView.
+            _myMapView.GraphicsOverlays.Add(_incidentGraphicsOverlay);
+            _myMapView.GraphicsOverlays.Add(_facilityGraphicsOverlay);
+        }
+
+        private void OnDrawStatusChanged(object sender, DrawStatusChangedEventArgs e)
+        {
+            if (e.Status == DrawStatus.Completed)
+            {
+                // Link the action of tapping on the map with the MyMapView_GeoViewTapped method.
+                _myMapView.GeoViewTapped += MyMapView_GeoViewTapped;
+
+                // Remove this method from DrawStatusChanged events.
+                _myMapView.DrawStatusChanged -= OnDrawStatusChanged;
+            }
+        }
+
+        private void MyMapView_GeoViewTapped(object sender, Esri.ArcGISRuntime.UI.Controls.GeoViewInputEventArgs e)
+        {
+            // Clear any prior incident and routes from the graphics.
+            _incidentGraphicsOverlay.Graphics.Clear();
+
+            // Create a MapPoint where the user clicked.
+            _incidentPoint = _myMapView.ScreenToLocation(e.Position);
+
+            // Populate the facility parameters than solve using the task.
+            PopulateParametersAndSolveRouteAsync();
+        }
+
+        private async void PopulateParametersAndSolveRouteAsync()
+        {
+            // Set facilities and incident in parameters.
+            ClosestFacilityParameters _closestFacilityParameters = await _task.CreateDefaultParametersAsync();
+            _closestFacilityParameters.SetFacilities(_facilities);
+            _closestFacilityParameters.SetIncidents(new List<Incident> { new Incident(_incidentPoint) });
+
+            try
+            {
+                // Use the task to solve for the closest facility.
+                ClosestFacilityResult result = await _task.SolveClosestFacilityAsync(_closestFacilityParameters);
+
+                // Get the index of the closest facility to incident. (0) is the index of the incident, [0] is the index of the closest facility.
+                int closestFacility = result.GetRankedFacilityIndexes(0)[0];
+
+                // Get route from closest facility to the incident and display to mapview.
+                ClosestFacilityRoute route = result.GetRoute(closestFacility, 0);
+
+                // Add graphics for the incident and route.
+                _incidentGraphicsOverlay.Graphics.Add(new Graphic(_incidentPoint, _incidentSymbol));
+                _incidentGraphicsOverlay.Graphics.Add(new Graphic(route.RouteGeometry, _routeSymbol));
+            }
+            catch (Esri.ArcGISRuntime.Http.ArcGISWebException exception)
+            {
+                if (exception.Message.ToString().Equals("Unable to complete operation."))
```

Comment:
The `ToString()` calls here and a few lines below aren't necessary because `Message` is already a string.